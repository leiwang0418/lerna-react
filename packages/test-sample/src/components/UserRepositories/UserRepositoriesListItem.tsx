import React, { FC, Fragment } from "react";
import PropTypes from "prop-types";
import {
	Divider,
	IconButton,
	ListItem,
	ListItemText,
	ListItemSecondaryAction,
} from "@material-ui/core";
import GitHubIcon from "@material-ui/icons/GitHub";
import { useIntl, defineMessage } from "react-intl";

export interface Props {
	name: string;
	description?: string;
	html_url: string;
}

const UserRepositoriesListItem: FC<Props> = ({
	name,
	description,
	html_url,
}) => {
	const intl = useIntl();

	return (
		<Fragment>
			<ListItem alignItems="flex-start">
				<ListItemText primary={name} secondary={description} />
				<ListItemSecondaryAction>
					<IconButton
						edge="end"
						aria-label={intl.formatMessage(message)}
						href={html_url}
						target="_blank"
					>
						<GitHubIcon />
					</IconButton>
				</ListItemSecondaryAction>
			</ListItem>
			<Divider component="li" />
		</Fragment>
	);
};

const message = defineMessage({
	defaultMessage: "前往github查看",
	description: "前往github查看",
});

UserRepositoriesListItem.propTypes = {
	name: PropTypes.string.isRequired,
	description: PropTypes.string,
	html_url: PropTypes.string.isRequired,
};

export default UserRepositoriesListItem;
