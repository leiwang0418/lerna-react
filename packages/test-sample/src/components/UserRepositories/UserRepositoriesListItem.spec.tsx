import { render } from "@test-utils";

import UserRepositoriesListItem, { Props } from "./UserRepositoriesListItem";

describe("UserRepositoriesListItem", () => {
	it("should render UserRepositoriesListItem", () => {
		const props: Props = {
			name: "test repository",
			html_url: "url",
		};

		const { container } = render(<UserRepositoriesListItem {...props} />);

		expect(container).toMatchSnapshot();
	});
});
