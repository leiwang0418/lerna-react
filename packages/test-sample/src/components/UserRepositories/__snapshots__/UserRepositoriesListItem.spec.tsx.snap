// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRepositoriesListItem should render UserRepositoriesListItem 1`] = `
<div>
  <li
    class="MuiListItem-container"
  >
    <div
      class="MuiListItem-root MuiListItem-gutters MuiListItem-alignItemsFlexStart MuiListItem-secondaryAction"
    >
      <div
        class="MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
        >
          test repository
        </span>
      </div>
    </div>
    <div
      class="MuiListItemSecondaryAction-root"
    >
      <a
        aria-disabled="false"
        aria-label="前往github查看"
        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
        href="url"
        tabindex="0"
        target="_blank"
      >
        <span
          class="MuiIconButton-label"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
            />
          </svg>
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </div>
  </li>
  <li
    class="MuiDivider-root"
    role="separator"
  />
</div>
`;
